src/mon/PGMap.cc

```diff
@@ -2527,16 +2528,21 @@ void PGMap::get_health_checks(
         summary = "Degraded data redundancy: ";
         sev = HEALTH_WARN;
         break;
-      case DEGRADED_FULL:
-        health_code = "PG_DEGRADED_FULL";
-        summary = "Degraded data redundancy (low space): ";
-        sev = HEALTH_ERR;
+      case BACKFILL_FULL:
+        health_code = "PG_BACKFILL_FULL";
+        summary = "Low space blocking backfill (add storage if this doesn't resolve itself): ";
+        sev = HEALTH_WARN;
         break;
       case DAMAGED:
         health_code = "PG_DAMAGED";
         summary = "Possible data damage: ";
         sev = HEALTH_ERR;
         break;
+      case RECOVERY_FULL:
+        health_code = "PG_RECOVERY_FULL";
+        summary = "Full OSDs blocking recovery: ";
+        sev = HEALTH_ERR;
```

Comment:
will RECOVERY_FULL never resolve itself?

Reply:
Line 2387: maps state to that: { PG_STATE_RECOVERY_TOOFULL, {RECOVERY_FULL, {}} },

Reply:
@jdurgin Recovery doesn't reserve storage for incoming data at reservation time.  Instead when any OSD goes over full ratio, recovery stops and the PG gets marked recover_toofull.  At that point unless client or other recovery are doing deletes, everything grinds to a halt.  Recovery can happen and client I/O is also blocked due to the full condition.  There could be some situations where it would resolve, but most likely additional space is needed or balancing must occur.  A user should make every attempt to never have OSDs running close to full.  The near full warning and backfill full setting are supposed to help prevent the situation.

Reply:
got it, sounds good as-is then